### PR TITLE
Fix: Block place sounds on mod platforms

### PR DESCRIPTION
**Pull request by onebeastchris**
Resolves https://github.com/GeyserMC/Geyser/issues/4823

### CODE REVIEW

File: bootstrap/mod/src/main/java/org/geysermc/geyser/platform/mod/mixin/server/ModBlockPlaceListener.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.cloudburstmc.math.vector.Vector3f;
+import org.geysermc.geyser.platform.mod.world.GeyserModWorldManager;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(BlockItem.class)
+public class ModBlockPlaceListener {
```
Comment (Konicai) [inline bootstrap/mod/src/main/java/org/geysermc/geyser/platform/mod/mixin/server/ModBlockPlaceListener.java:47]: apart from the vast majority of cases where a class implements/extends something from the `core` package, i don't think its necessary to place the platform name at the front

just me tho

File: core/src/main/java/org/geysermc/geyser/translator/protocol/java/level/JavaBlockUpdateTranslator.java [hunk 2]
```diff
@@ -52,15 +52,15 @@ public void translate(GeyserSession session, ClientboundBlockUpdatePacket packet
         this.checkInteract(session, packet);
     }
 
-    private boolean checkPlace(GeyserSession session, ClientboundBlockUpdatePacket packet) {
+    private void checkPlace(GeyserSession session, ClientboundBlockUpdatePacket packet) {
```
Comment (Konicai) [inline core/src/main/java/org/geysermc/geyser/translator/protocol/java/level/JavaBlockUpdateTranslator.java:55]: what do you think of `checkPlaceSound`?

File: bootstrap/mod/src/main/java/org/geysermc/geyser/platform/mod/world/GeyserModWorldManager.java [hunk 3]
```diff
@@ -58,16 +62,19 @@
...
 public class GeyserModWorldManager extends GeyserWorldManager {
 
     private static final GsonComponentSerializer GSON_SERIALIZER = GsonComponentSerializer.gson();
     private final MinecraftServer server;
+    private static GeyserImpl geyser;
 
-    public GeyserModWorldManager(MinecraftServer server) {
+    public GeyserModWorldManager(GeyserImpl geyser, MinecraftServer server) {
         this.server = server;
+        GeyserModWorldManager.geyser = geyser;
```
Comment (Konicai) [inline bootstrap/mod/src/main/java/org/geysermc/geyser/platform/mod/world/GeyserModWorldManager.java:77]: at this point we may as well just move `handleBlockPlace` into the mixin and use GeyserImpl#getInstance lol

Comment (onebeastchris) [reply bootstrap/mod/src/main/java/org/geysermc/geyser/platform/mod/world/GeyserModWorldManager.java:77]: sure, will do that

File: core/src/main/java/org/geysermc/geyser/translator/protocol/java/level/JavaBlockUpdateTranslator.java [hunk 4]
```diff
@@ -104,4 +103,9 @@ private void checkInteract(GeyserSession session, ClientboundBlockUpdatePacket p
         session.setInteracting(false);
         BlockSoundInteractionTranslator.handleBlockInteraction(session, lastInteractPos.toFloat(), state);
     }
+
+    // Spigot simply listens for the block place event, and then there's mixins
+    private boolean handledByPlatform(PlatformType type) {
+        return type == PlatformType.SPIGOT || type == PlatformType.NEOFORGE || type == PlatformType.FABRIC;
+    }
```
Comment (Konicai) [inline core/src/main/java/org/geysermc/geyser/translator/protocol/java/level/JavaBlockUpdateTranslator.java:110]: isn't this equivalent to `WorldManager#hasOwnChunkCache`?

**Review by Camotoy — APPROVED** [review]

**Pull request merged**